ClinGen/clincoded
# Connects to #954. Ability to associate PMID in VCI

![image](https://cloud.githubusercontent.com/assets/4326866/18619862/b93d8350-7dbb-11e6-8c94-a7750b7888bf.png)

* Added in ability to add PMID evidence to an `interpretation` object via the new `extra_evidence` object
* Added in a module to display this evidence list in the VCI (multiple times per page if needed)
  * Module also supports in-table modal popup to add the articles (via the `add_external_resource` modal) and in-table editing and deleting of `extra_evidence` objects

* The `interpretation` object links to `extra_evidence` via the `extra_evidence_list array`
* The `extra_evidence` object currently only links to `articles`, but can theoretically support different objects such as `families`, `groups`, etc., as well.

**Note:** the article link from `extra_evidence` is currently an array to support multiple articles per evidence object, but the UI only allows for one article per evidence object. Whether or not the user will be allowed to add more than one articles per evidence is still under debate and may change in the future.

Testing:

1. Get to VCI
2. Go to Experimental and Segregation/Case tabs and confirm that there are no extra evidence tables
3. Begin interpretation, then confirm that the extra evidence tables are visible in the two tabs
4. Add PMID evidence
  * Also confirm that the Save button cannot be pressed when Evidence text box is empty
  * Also confirm that the Cancel button clears the returned PMID and hides the Evidence text box
  * Also confirm that pressing the 'Edit PMID' button brings up the modal again and can change the retrieved article
5. Add another PMID evidence
6. Add another PMID evidence, but with the same PMID as a previous evidence
7. Edit one of the previously added evidence items
  * Also confirm that pressing Edit on another evidence item switches the Edit mode to the new item
  * Also confirm that the Cancel button cancels out of Edit mode
8. Delete one of the previously added evidence items
